Sources/NIOCore/ChannelPipeline.swift

```diff
@@ -2089,3 +2091,171 @@ extension ChannelPipeline: CustomDebugStringConvertible {
         return handlers
     }
 }
+
+extension ChannelPipeline {
+    private enum AuditDirection: Equatable {
+        case inbound
+        case outbound
+    }
+
+    /// Audit the total number of bytes buffered for outbound.
+    public func auditOutboundBufferedBytes() -> EventLoopFuture<Int> {
+        let future: EventLoopFuture<Int>
+
+        if self.eventLoop.inEventLoop {
+            future = self.eventLoop.makeSucceededFuture(auditAll(direction: .outbound))
+        } else {
+            future = self.eventLoop.submit {
+                self.auditAll(direction: .outbound)
+            }
+        }
+
+        return future
+    }
+    
+    /// Audit the total number of bytes buffered for inbound.
+    public func auditInboundBufferedBytes() -> EventLoopFuture<Int> {
+        let future: EventLoopFuture<Int>
+
+        if self.eventLoop.inEventLoop {
+            future = self.eventLoop.makeSucceededFuture(auditAll(direction: .inbound))
+        } else {
+            future = self.eventLoop.submit {
+                self.auditAll(direction: .inbound)
+            }
+        }
+
+        return future
+    }
+    
+    /// Audit the total numbers of outbound bytes buffered in the channel handler with the given `name`.
+    ///
+    /// - parameters:
+    ///     - name: the name of the channel handler whose outbound buffered bytes will be audited.
+    public func auditOutboundBufferedBytes(name: String) -> EventLoopFuture<Int> {
```

Comment:
I'm not sure this API is pulling its weight: users can always obtain channel handlers by name and call the relevant functions directly.